### DISCUSSION TITLE
Can I join the same table?

### DISCUSSION THREAD

**Discussion by myyrakle** [diesel-rs/diesel#2739] [Q&A] [answered]
The table has a key value that refers to itself, so you want to join the same table.

Is this a function currently available?
If possible, which method should you use?

example
```
let history_list: Vec<_> = tb_document_history::table // as A
            .inner_join(tb_user::table.on(tb_user::dsl::id.eq(tb_document_history::dsl::writer_id))) // as B
            .left_join(
                tb_document_history::table // as C?
                    .on(tb_document_history::dsl::id.eq(tb_document_history::dsl::rollback_id)
                ),
            )
            .filter(tb_document_history::dsl::document_id.eq(document.id))
            //...
```

**Comment by weiznich** [comment, marked as answer]
You would need to provide your own alias extension to the query builder, otherwise this is not supported yet using the built-in dsl. https://github.com/diesel-rs/diesel/issues/1611 has some details on possible solutions.

**Reply by myyrakle** [reply]
Thank you for answer.
